module4/homework/appendNewRecipe/AppendNewRecipe.cpp

```diff
@@ -0,0 +1,62 @@
+#include "AppendNewRecipe.hpp"
+
+#include <algorithm>
+#include <fstream>
+
+bool AppendNewRecipe(std::vector<std::string> steps,
+                     const std::list<std::string>& ingredients,
+                     const std::deque<std::pair<size_t, char>>& amount) {
+    std::ofstream file("recipes.txt", std::ios::app);
+    if (file.is_open()) {
+        auto text = FormatRecipit(steps, ingredients, amount).str();
+        file << text;
+        file.close();
+        return true;
+    }
+    return false;
+}
+
+std::vector<std::string> FormatIngredients(const std::list<std::string>& ingredients,
+                                           const std::deque<std::pair<size_t, char>>& amount) {
+    if (ingredients.size() != amount.size()) {
+        return {};
+    }
+    std::vector<std::string> formattedIngredients(ingredients.size());
+    std::transform(begin(ingredients),
+                   end(ingredients),
+                   begin(amount),
+                   begin(formattedIngredients),
+                   [](const auto& ingredient, const auto& pair) {
+                       auto [amount, unit] = pair;
+                       std::string formattedUnit{};
+                       if (unit == 'g') {
+                           formattedUnit = "gram";
+                       }
+                       if (unit == 's') {
+                           formattedUnit = "szklanka(i)";
+                       }
+                       if (unit == 'm') {
+                           formattedUnit = "mililitrow";
+                       }
+                       return std::to_string(amount) + ' ' + formattedUnit + ' ' + ingredient;
+                   });
+    return formattedIngredients;
+}
+
+std::stringstream FormatRecipit(std::vector<std::string> steps,
+                                const std::list<std::string>& ingredients,
+                                const std::deque<std::pair<size_t, char>>& amount) {
+    std::stringstream stream;
+    stream << "Skladniki:\n";
```

Comment:
I suggest to make these sentences (Skladniki, Kroki, etc) as const or constexpr variables.

Reply:
+1

Reply:
+1